src/Integration.Vsix.UnitTests/CFamily/CFamilyHelperTests.cs

```diff
@@ -39,21 +39,74 @@ public class CFamilyHelperTests
     {
 
         [TestMethod]
-        public void CreateRequest_HeaderFile_IsNotProcessed()
+        public void CreateRequest_HeaderFile_IsSupported()
         {
             // Arrange
             var loggerMock = new Mock<ILogger>();
-
-            var projectItemMock = new Mock<ProjectItem>();
+            ProjectItemConfig projectItemConfig = new ProjectItemConfig();
+            projectItemConfig.itemType = "ClInclude";
+            var rulesConfig = GetDummyRulesConfiguration();
             var rulesConfigProviderMock = new Mock<ICFamilyRulesConfigProvider>();
+            rulesConfigProviderMock
+                .Setup(x => x.GetRulesConfiguration(It.IsAny<string>()))
+                .Returns(rulesConfig);
+            var projectItemMock = CreateProjectItemWithProject("c:\\foo\\xxx.vcxproj", projectItemConfig);
 
             // Act
             var request = CFamilyHelper.CreateRequest(loggerMock.Object, projectItemMock.Object, "c:\\dummy\\file.h",
                 rulesConfigProviderMock.Object, null);
 
             // Assert
-            AssertMessageLogged(loggerMock, "Cannot analyze header files. File: 'c:\\dummy\\file.h'");
-            request.Should().BeNull();
+            loggerMock.Verify(x => x.WriteLine(It.IsAny<string>()), Times.Never);
```

Comment:
IMO it's better not to check the logger - if we were to add random logging in the future, this test would break for no real reason. The interesting bit is that the request is not null.